libgalois/include/katana/DynamicBitset.h

```diff
@@ -86,41 +86,49 @@ class KATANA_EXPORT DynamicBitset {
    *
    * @param n Size to reserve the capacity of the bitset to
    */
-  void reserve(uint64_t n) {
+  void reserve(size_t n) {
     KATANA_LOG_DEBUG_ASSERT(
-        bits_uint64 == 64);  // compatibility with other devices
-    bitvec.reserve((n + bits_uint64 - 1) / bits_uint64);
+        kNumBitsInUint64 == 64);  // compatibility with other devices
+    bitvec_.reserve((n + kNumBitsInUint64 - 1) / kNumBitsInUint64);
   }
 
   /**
    * Clears the bitset.
    */
   void clear() {
-    num_bits = 0;
-    bitvec.clear();
+    num_bits_ = 0;
+    bitvec_.clear();
   }
 
   /**
    * Shrinks the allocation for bitset to its current size.
    */
-  void shrink_to_fit() { bitvec.shrink_to_fit(); }
+  void shrink_to_fit() { bitvec_.shrink_to_fit(); }
+
+  /**
+   * Swaps the bitset with the given one
+   */
+  void swap(DynamicBitset& bitset) {
```

Comment:
Implementing `swap` directly is a slightly older pattern and more common pre C++11. If you did go this path, technically we should satisfy https://en.cppreference.com/w/cpp/types/is_swappable , which would require a free function `swap(DynamicBitset& a, DynamicBitset& b)`.

Easier to just implement move semantics (constructor and assignment) and not `swap` directly. `std::swap` will then just pick up on the moveability and do what you want.

Reply:
Ok, I'll add a move constructor instead of `swap`.